zipline/finance/performance/tracker.py

```diff
@@ -149,6 +157,7 @@ def __init__(self, sim_params, trading_calendar, env):
             # initial cash is your capital base.
             starting_cash=self.capital_base,
             data_frequency=self.sim_params.data_frequency,
+            portfolio=portfolio,
```

Comment:
Having looked through this code some more, I don't think we ever inspect to call `as_portfolio` on the daily performance period; in fact, it would be a serious bug to do so, because that method mutates the shared portfolio object in place.  Given that that's the case, I think what we should do here is pass a dummy portfolio object that raises an error if we try to do anything with it.

Longer term, I think we should either merge the cumulative and daily performance periods together into one object or break them out into separate classes.  As things currently stand, they're both instances of the same `PerformancePeriod` class, but they behave and are used very differently.

Reply:
I think @ssanderson meant to make this comment for `self.todays_performance` below, where I've changed the portfolio to be `portfolio=DummyPortfolio()`. For `self.cumulative_performance` here I think we still need the actual portfolio.